python/shotgun_model/shotgun_hierarchy_model.py

```diff
@@ -255,6 +267,30 @@ def canFetchMore(self, index):
     ############################################################################
     # protected methods
 
+    def _check_hierarchy_is_supported(self):
+        """
+        Checks the current Shotgun connection to make sure it supports
+        hierarchy queries.
+        """
+
+        # get the current shotgun connection
+        current_engine = sgtk.platform.current_engine()
+        sg_connection = current_engine.shotgun
+        server_caps = sg_connection.server_caps
```

Comment:
Edited original response as it had a bad test case example.

If I call `nav_expand` with a newer version of python-api on old site, I get:

```python
>>> shotgun.server_caps.version
(6, 3, 9)
>>> shotgun.nav_expand("/", "Version.entity")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/shotgun/configs/sgutils_test/install/core/python/tank_vendor/shotgun_api3/shotgun.py", line 2068, in nav_expand
    'entity_fields': entity_fields
  File "/shotgun/configs/sgutils_test/install/core/python/tank/authentication/shotgun_wrapper.py", line 59, in _call_rpc
    return super(ShotgunWrapper, self)._call_rpc(*args, **kwargs)
  File "/shotgun/configs/sgutils_test/install/core/python/tank_vendor/shotgun_api3/shotgun.py", line 2149, in _call_rpc
    self._response_errors(response)
  File "/shotgun/configs/sgutils_test/install/core/python/tank_vendor/shotgun_api3/shotgun.py", line 2425, in _response_errors
    raise Fault(sg_response.get("message", "Unknown Error"))
Fault: Uncaught exception! undefined method `nav_expand' for #<API3Handler:0x007f166622aca8>
>>> 
```

I agree, it feels like a proper interface or expected exception for checking supported features seems right. Like I said before, I've added to hot topics for discussion.